kroxylicious/src/main/java/io/kroxylicious/proxy/internal/clusterendpointprovider/SniAwareClusterEndpointConfigProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusterendpointprovider;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import io.kroxylicious.proxy.config.BaseConfig;
+import io.kroxylicious.proxy.service.ClusterEndpointConfigProvider;
+import io.kroxylicious.proxy.service.HostPort;
+
+public class SniAwareClusterEndpointConfigProvider implements ClusterEndpointConfigProvider {
```

Comment:
The name is a bit poor. `SniRouting` or  `SniRouterClusterEndpointConfigProvider` would probably be better.

Reply:
I went with SniRouting.